.github/workflows/changelog.yml

```diff
@@ -6,11 +6,32 @@ on:
 permissions: "read-all"
 
 jobs:
+  # Check for label in its own job to allow skipping multiple steps with 'success'
+  check-labels:
+    name: check labels
+    runs-on: ubuntu-latest
+    steps:
+      - id: check-labels
+        env:
+          OWNER: ${{ github.repository_owner }}
+          REPO_NAME: ${{ github.event.repository.name }}
+          PR_ID: ${{ github.event.pull_request.number }}
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          labels="$(gh api repos/$OWNER/$REPO_NAME/pulls/$PR_ID --jq '.labels.[].name')"
+          if [[ $labels != *"skip changelog"* ]]; then
+            echo "needs_entry=true" >> "$GITHUB_OUTPUT"
+          else
+            echo "needs_entry=false" >> "$GITHUB_OUTPUT"
+          fi
+    outputs:
+      needs_entry: ${{ steps.check-labels.outputs.needs_entry }}
+
   check-changelog-entry:
-    name: changelog entry
+    name: has fragment
```

Comment:
Can we make the name clearer for this? Like "has changelog fragment"?

Reply:
works for me